### PR TITLE
refactor(db access) - Extract direct db access from fxa-graphql-api to fxa-shared

### PR DESCRIPTION
**Pull request by jackiemunroe**
## Because

- We want to get away from more stored procedures

## This pull request

- Extracts direct db access out of fxa-graphql-api and moved it to fxa-shared

## Issue that this pull request solves

Closes: #6338

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).

### CODE REVIEW

**Comment by meandavejustice** [comment]
oof-marone

This is a big pr! happy to see this happening, ping me when ur ready for review so we can get a couple eyes on it

File: packages/fxa-shared/config.ts [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+const configDir = path.dirname(path.dirname(__dirname));
+let envConfig = path.join(configDir, 'config', `${conf.get('env')}.json`);
+envConfig = `${envConfig},${process.env.CONFIG_FILES || ''}`;
+const files = envConfig.split(',').filter(fs.existsSync);
+conf.loadFile(files);
+conf.validate({ allowed: 'strict' });
+const Config = conf;
+
+export type AppConfig = typeof Config;
+export default Config;
```
Comment (bbangert) [inline packages/fxa-shared/config.ts:211]: I think you moved too much in this file. I was thinking about this earlier today as well, and I think for shared components like this, what makes sense would be to have the shared component include either the config or config generator that the package using it should import and use in its local config.

For example, in this case, the docs for the `fxa-shared-db/lib/db` (a README in its dir), could say something like:

> To use the auth db in your package, import `makeMySQLConfig` from `fxa-shared-db/lib/db/config`, and put it in your packages convict setup. Then pass the config into `setupAuthDatabase`.

So only the bits needed to ease proper use would be included.

File: packages/fxa-shared/lib/constants.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import { Token } from 'typedi';
+import { AppConfig } from '../config';
+import { Logger } from 'mozlog';
+import { RedisConfig } from '../config';
+import AuthClient from 'fxa-auth-client';
+
+export const configContainerToken = new Token<AppConfig>();
+export const loggerContainerToken = new Token<Logger>();
+
+export const fxAccountClientToken = new Token<AuthClient>();
```
Comment (bbangert) [inline packages/fxa-shared/lib/constants.ts:14]: These aren't relevant to the database.

File: packages/fxa-shared/lib/constants.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+export const authRedisConfig = new Token<RedisConfig>();
```
Comment (bbangert) [inline packages/fxa-shared/lib/constants.ts:15]: Oh, neither is this. The database actually doesn't need any container setup.

**Comment by jackiemunroe** [comment]
replaced by https://github.com/mozilla/fxa/pull/6495

**Pull request closed without merging**